### ISSUE TITLE
CSRF issue

### ISSUE THREAD

**Issue by IronicBadger** [linuxserver/docker-healthchecks#22]
Linked to upstream issue [#239](https://github.com/healthchecks/healthchecks/issues/239).

```
Forbidden (403)
CSRF verification failed. Request aborted.

Reason given for failure:

    Referer checking failed - https://domain.name.tld/accounts/login/ does not match any trusted origins.
```

**Comment by Cheezzhead** [comment]
I managed to fix it by manipulating the Host header that nginx passes to Django. According to [this](https://docs.djangoproject.com/en/2.2/ref/settings/#csrf-trusted-origins), the Referer header has to match the origin present in the Host header.

I (and I assume many others) am using the `linuxserver/letsencrypt` with the built-in reverse proxy configurations, and was using the `/config/nginx/proxy.conf` file, which sends this Host header:

`proxy_set_header Host $host:$server_port;`

I changed this header for my healthchecks subdomain to `proxy_set_header Host $host;`, after which the CSRF verification failed error went away.


Everybody's config may be different, but I just wanted to let people know that if you have the above `proxy_set_header` line somewhere in your nginx config, removing the $server_port might resolve your issue.

**Comment by aptalca** [comment]
That's really silly that the addition of the port breaks CSRF

**Comment by joshuaavalon** [comment]
If you are using reverse proxy (I am using Traefik), adding `USE_X_FORWARDED_HOST = True` fix the problem.

**Comment by alex-phillips** [comment]
django also has the ability to add the following in the `local_settings.py` file:

```
CSRF_TRUSTED_ORIGINS = ['healthchecks.yourdomain.com']
```

**Comment by aptalca** [comment]
Fixed in #32
